### PR TITLE
input_common: Implement official GameCube adapter support

### PR DESCRIPTION
**Pull request by ameerj**
This PR aims to add the Nintendo GameCube adapter as a new input device. It's a port of the implementation I initially added into yuzu-emu, along with the later code and feature improvements by @german77 and @lioncash 

Although GameCube adapter support may not be as highly requested on Citra as other emulators, it could still be a convenient addition for those who do wish to use this device. 

To use the controller, [follow the same process as you would for yuzu/Dolphin](https://yuzu-emu.org/wiki/faq/#how-do-i-use-my-gamecube-controller-adapter).

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/5735)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -193,6 +193,17 @@ if (ENABLE_QT)
     endif()
 endif()
 
+# Ensure libusb is properly configured (based on dolphin libusb include)
+if(NOT APPLE)
+    include(FindPkgConfig)
+    find_package(LibUSB)
```
Comment (jbeich) [inline CMakeLists.txt:199]: `FindLibUSB.cmake` is missing in Citra compared to Yuzu. As FreeBSD (and DragonFly) uses [different libusb implementation](https://github.com/freebsd/freebsd-src/tree/releng/13.0/lib/libusb) this breaks build. From [error log](https://github.com/citra-emu/citra/files/6113571/citra-qt5-s20210308.log):
```
[...]
-- Found PkgConfig: pkgconf (found version "1.7.3") 
CMake Warning at CMakeLists.txt:198 (find_package):
  By not providing "FindLibUSB.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "LibUSB", but
  CMake did not find one.

  Could not find a package configuration file provided by "LibUSB" with any
  of the following names:

    LibUSBConfig.cmake
    libusb-config.cmake

  Add the installation prefix of "LibUSB" to CMAKE_PREFIX_PATH or set
  "LibUSB_DIR" to a directory containing one of the above files.  If "LibUSB"
  provides a separate development package or SDK, be sure it has been
  installed.
[...]
FAILED: bin/Release/citra-qt 
ld: error: undefined symbol: usbi_backend
>>> referenced by core.c
>>>               core.c.o:(usbi_alloc_device) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(usbi_alloc_device) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_has_capability) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_has_capability) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(usbi_connect_device) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(usbi_disconnect_device) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_get_device_list) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_get_device_list) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_unref_device) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_unref_device) in archive externals/libusb/libusb.a
>>> referenced 46 more times

ld: error: undefined symbol: usbi_write
>>> referenced by core.c
>>>               core.c.o:(usbi_signal_event) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_close) in archive externals/libusb/libusb.a

ld: error: undefined symbol: usbi_read
>>> referenced by core.c
>>>               core.c.o:(usbi_clear_event) in archive externals/libusb/libusb.a
>>> referenced by core.c
>>>               core.c.o:(libusb_close) in archive externals/libusb/libusb.a

ld: error: undefined symbol: usbi_close
>>> referenced by io.c
>>>               io.c.o:(usbi_io_init) in archive externals/libusb/libusb.a
>>> referenced by io.c
>>>               io.c.o:(usbi_io_init) in archive externals/libusb/libusb.a
>>> referenced by io.c
>>>               io.c.o:(usbi_io_exit) in archive externals/libusb/libusb.a
>>> referenced by io.c
>>>               io.c.o:(usbi_io_exit) in archive externals/libusb/libusb.a

ld: error: undefined symbol: usbi_inc_fds_ref
>>> referenced by io.c
>>>               io.c.o:(handle_events) in archive externals/libusb/libusb.a

ld: error: undefined symbol: usbi_poll
>>> referenced by io.c
>>>               io.c.o:(handle_events) in archive externals/libusb/libusb.a

ld: error: undefined symbol: usbi_dec_fds_ref
>>> referenced by io.c
>>>               io.c.o:(handle_events) in archive externals/libusb/libusb.a
c++: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by ameerj** [comment]
@jbeich thanks for the comment. I missed your unbreak commit when going back through the yuzu PRs relating to GC. I added the `FindLibUSB.cmake` commit. Please let me know if this addition resolves the BSD build failure.

**Comment by jbeich** [comment]
Thanks, [builds fine](https://github.com/citra-emu/citra/files/6118844/citra-qt5-s20210310.log) now.

**Comment by Morph1984** [comment]
Might want to port this too https://github.com/yuzu-emu/yuzu/pull/6078

**Comment by kylon** [comment]
this breaks build in ArchLinux, libusb has a different path

File: src/input_common/CMakeLists.txt [hunk 2]
```diff
@@ -30,3 +34,4 @@ endif()
 
 create_target_directory_groups(input_common)
 target_link_libraries(input_common PUBLIC core PRIVATE common ${Boost_LIBRARIES})
+target_link_libraries(input_common PUBLIC ${LIBUSB_LIBRARIES})
```
Comment (lat9nq) [inline src/input_common/CMakeLists.txt:37]:
```suggestion
target_include_directories(input_common PRIVATE ${LIBUSB_INCLUDE_DIR})
target_link_libraries(input_common PUBLIC ${LIBUSB_LIBRARIES})
```
Need to tell CMake where to look for `libusb.h`. Fixes https://github.com/citra-emu/citra/pull/5735#issuecomment-826104156

**Comment by ameerj** [comment]
@kylon can you confirm if building errors on Arch are resolved now?

**Comment by kylon** [comment]
> @kylon can you confirm if building errors on Arch are resolved now?

yeah, sorry

yes, it compiles now (not the AUR package because it still pulls an old version), so Arch build is fixed

thanks

**Review by bunnei — APPROVED** [review]

**Comment by jbeich** [comment]
a80e5664645a reverted b660adea0f84 and a2f9fef97e09. Likely a rebase mistake as 57fc1633f301 didn't touch `externals/`.

**Comment by ameerj** [comment]
@jbeich I completely missed that. Thanks for catching it. Mind opening a PR updating catch and dynarmic and i can merge it.

**Pull request merged**